optica.rb

```diff
@@ -68,15 +68,28 @@ class Optica < Sinatra::Base
       halt(403) unless ip == header.split(',').first
     end
 
+    # update main store
     begin
       merged_data = settings.store.add(ip, data)
-      settings.events.send(merged_data.merge("event" => data))
     rescue
       halt(500)
     end
 
+    # publish update event
+    error_msg = ""
+    begin
+      settings.events.send(merged_data.merge("event" => data))
+    rescue Exception => e
```

Comment:
it's probably a mistake to rescue `Exception` -- we shouldn't be rescuing any exception class above `StandardError`. this might contribute to problems exiting the application.